.github/workflows/csv-to-csvw.yml

```diff
@@ -52,52 +53,68 @@ jobs:
             file_name="${file_path##*/}"
             file_extension="${file##*.}"
             out_path="out/${file_path}/"
-            
+            root_path=$(echo "$file_path" | cut -d "/" -f1)
+
             echo "file_path: ${file_path}"
             echo "file_name: ${file_name}"
             echo "file_extension: ${file_extension}"
             echo "out_path: ${out_path}"
+            echo "root_path: ${root_path}"
 
-            if [[ $file_extension == "csv" ]]; then
+            if [[ ($file_extension == "csv" || $file_extension == "json") && $root_path != "out" ]]; then
               echo $'\n'
               echo "Processing file ${file}"
               echo "Output path is ${out_path}"
 
-              csv_file=${file}
-              for file_secondary in "${added_modified_files[@]}"; do
-                file_secondary_name="${file_secondary%.*}"
-                file_secondary_extension="${file_secondary##*.}"
-                if [[ "${file_secondary_name}.${file_secondary_extension}" == "${file_path}.json" ]]; then
-                  config_file="${file_secondary_name}.json"
+              csv_file=""
+              config_file=""
+              if [[ $file_extension == "csv" ]]; then
+                csv_file=${file}
+                for file_secondary in "${added_modified_files[@]}"; do
+                  file_secondary_name="${file_secondary%.*}"
+                  file_secondary_extension="${file_secondary##*.}"
+                  if [[ "${file_secondary_name}.${file_secondary_extension}" == "${file_path}.json" ]]; then
+                    config_file="${file_secondary_name}.json"
+                  fi
+                done
+              elif [[ $file_extension == "json" ]]; then
+                config_file=${file}
+                if [[ -f "${file_path}/${file_name}.csv" ]]; then
+                  csv_file="${file_path}/${file_name}.csv"
                 fi
```

Comment:
If we're not able to find a matching CSV file, should we set `config_file=NULL` or similar just for clarity that it isn't really a configuration file for a CSV?

Reply:
Yes good shout - I have updated.